include/coap2/subscribe.h

```diff
@@ -17,23 +17,49 @@
 #define COAP_SUBSCRIBE_H_
 
 /**
- * @defgroup observe Resource Observation
+ * @defgroup observe Observe Subscription
```

Comment:
Hm, "Resource Observation" is the term used by RFC 7641. I think it more appropriate to name the section after the feature's official name.

Reply:
Reverted back